samples: dect: add simple sample (#136)
lemrey/sdk-nrf: samples/dect/dect_phy/hello_dect/README.rst

.. _nrf_modem_dect_phy_hello:

nRF91x1: DECT NR+ PHY hello sample
##################################

.. contents::
   :local:
   :depth: 2

This sample demonstrates how to set up a simple DECT NR+ application with the DECT PHY modem firmware.

> 

Reviewer: Looks like there are a few extra newlines here.

Requirements
************

> The sample needs at least two nRF91x1 Development Kits.

Reviewer: ```suggestion
This sample needs at least two nRF91x1 Development Kits.
```

See :ref:`nrf91x1_ug_intro` for more details.

.. note::

>    The :ref:`nrf_modem_dect_phy_simple` requires the DECT NR+ PHY modem firmware to run on the nRF91x1 modem core. Please contact our sales department to know more.

Reviewer: Perhaps a link to the contact would be useful to the user? Not sure how this has been approached before in NCS.

> 

Reviewer: Extra newline?

>    It is the responsibility of the developer to operate the devices according to the local regulation, both at the development site and the device operation regions.

Author: I'll move this to the building and running part and merge with the note there.

.. include:: /includes/tfm.txt

Overview
********

> The sample shows a simple broadcast and reception of DECT NR+ messages between devices on a hard-coded channel.

Reviewer: Reads better

```suggestion
This sample shows a simple broadcast and reception of DECT NR+ messages between devices on a hard-coded channel.
```

Author: `The sample` is used elsewhere.

After initialization the devices will run a loop, transmitting a counter value before listening for incoming receptions. The time to listen is given by the :kconfig:option:`CONFIG_RX_PERIOD_S` Kconfig option.
The loop is exited after a number of transmissions given by the :kconfig:option:`CONFIG_TX_TRANSMISSIONS` Kconfig option, or continue forever if the :kconfig:option:`CONFIG_TX_TRANSMISSIONS` Kconfig option is ``0``.

Each device can be reset to run the sample again.

Building and running
********************

.. |sample path| replace:: :file:`samples/modem_trace_flash`

.. include:: /includes/build_and_run_ns.txt

.. important::

>    DECT NR+ operates on free but regulated radio channels. The regulation and availability of the channels varies in different countries.

Reviewer: "varies by country"

>    It is your responsibility to operate the devices according to the local regulation, both at the development site and the device operation regions.

Reviewer: "and the device operating regions" perhaps

   If you are in the EU and US, you can use the ``overlay-eu.conf`` and ``overlay-us.conf``Kconfig overlays respectively.
   Otherwise set the carrier using the :kconfig:option:`CONFIG_CARRIER` Kconfig option.

Testing
=======

|test_sample|

> After programming the sample to your development kits, test it by performing the following steps:

Reviewer: ```suggestion
After programming the sample to your Development kits, test it by performing the following steps:
```

#. |connect_kit|
#. |connect_terminal|

> #. Observe that the devices will transmit a counter value that is received by the other devices.

Reviewer: Is this sentence correct or should it be:

```suggestion
#. Observe that the device will transmit a counter value that is received by the other devices.
```

Author: It should be plural.

.. _nrf_modem_dect_phy_hello:

nRF91x1: DECT NR+ PHY hello sample
##################################

.. contents::
   :local:
   :depth: 2

This sample demonstrates how to set up a simple DECT NR+ application with the DECT PHY modem firmware.

.. important::

   DECT NR+ operates on free but regulated radio channels. The regulation and availability of the channels varies in different countries.
   It is the responsibility of the developer to operate the devices according to the local regulation, both at the development site and the device operation regions.

Requirements
************

The sample needs at least two nRF91x1 Development Kits.

See :ref:`nrf91x1_ug_intro` for more details.

.. note::

   The :ref:`nrf_modem_dect_phy_simple` require a special modem firmware to run on the nRF91x1 modem core. Please contact our sales department to know more.

.. include:: /includes/tfm.txt

Overview
********

The sample shows a simple broadcast of DECT NR+ messages from one device to other devices on a hard-coded channel.

> During initialization the devices listens on the channel for broadcast transmissions.

Reviewer: ```suggestion
During initialization the devices listen on the channel for broadcast transmissions.
```

> Dependent on the detection of transmissions, the device takes one of the following roles:

Reviewer: ```suggestion
Depending on the detection of transmissions, the device takes one of the following roles:
```

> * If no transmission is detected, the device takes the role as a tranceiver. In this role, the device sends an incrementing counter value on the hard-coded channel. The period between each transmission is decided by the :kconfig:option:`CONFIG_TX_INTERVAL_SECONDS` Kconfig option.

Reviewer: ```suggestion
* If no transmission is detected, the device takes the role of a tranceiver. In this role, the device sends an incrementing counter value on the hard-coded channel. The period between each transmission is decided by the :kconfig:option:`CONFIG_TX_INTERVAL_SECONDS` Kconfig option.
```

> * If a transmission is detected, the device takes the role as a receiver. In this role, the device listens for incomming messages.

Reviewer: ```suggestion
* If a transmission is detected, the device takes the role of a receiver. In this role, the device listens for incoming messages.
```
